src/instructlab/utils.py

```diff
@@ -900,3 +914,73 @@ def list_models(
 def contains_argument(prefix: str, args: typing.Iterable[str]) -> bool:
     # Either --foo value or --foo=value
     return any(s == prefix or s.startswith(prefix + "=") for s in args)
+
+
+def get_model_arch(model_path: pathlib.Path) -> str:
+    """
+    Extract a given model's architecture from its config if available and return it
+
+    args
+        model_path (str): Path to the model, used to read the model config if available
+    returns
+        model_arch (str): The architecture of the model as expressed in the config file
+    """
+
+    # TODO: implement for GGUF models
```

Comment:
we may want to note what the blocker here is as part of this comment

Reply:
the blocker is just that I dont know how to do it yet - but plan to raise a follow up PR for it
im not sure documenting this will be helpful

Reply:
Isn't the blocker that we can't upgrade llama-cpp? or am i mistaken?

Reply:
that's not a blocker if you just want to extract the architecture of a GGUF model from it's metadata, I don't think

it is a blocker if you want to serve and chat with the model

Reply:
ack!